src/App.jsx

```diff
@@ -8,24 +8,39 @@ import { scrollToTop } from './helpers/scrollToTop';
 // Third party libraries
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem("savedTodoList")) ?? []);
-
-  useEffect(()=>{
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-};
-
-
 const App = () => {
   /*
     ============================
     =           HOOKS          =
     ============================
   */
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  /* Insert todoList to app from localStorage*/
+  useEffect(() => {
+    const todoListFetch = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem("savedTodoList")) ?? []
+          }
+        })
+      }, 2000);
+    });
+
+    todoListFetch.then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+    });
```

Comment:
Great job! :1st_place_medal: 

Another way of writing this is just creating a new promise without saving it into a variable
![image](https://user-images.githubusercontent.com/8095975/235016246-2b7a90c3-8c2c-46b7-9371-270359fc57d0.png)

Reply:
Thanks for the advice! It saves lines of code by doing this. I will update my code to your suggestion!